index.ts

```diff
@@ -1,30 +1,62 @@
-// @flow
-
 import GeocoderArcGIS from 'geocoder-arcgis'
-import lonlat from '@conveyal/lonlat'
+import type { GeoJSON } from 'geojson'
+import { toCoordinates, toString, toPoint } from '@conveyal/lonlat'
+import type { LonLatInput } from '@conveyal/lonlat'
 
-const arcGisGeocoders = {}
+type Point = {
+  x: number
+  y: number
+}
+
+type GeocodersList = {
+  [key: string]: GeocoderArcGIS
+}
 
 type Boundary = {
   rect: {
-    minLat: number,
-    minLon: number,
-    maxLat: number,
+    minLat: number
+    minLon: number
+    maxLat: number
     maxLon: number
   }
 }
 
+type Candidate = {
+  location: LonLatInput
+  attributes: Record<string, string | number>
+}
+
+type Options = {
+  searchExtent?: string
+  location?: string
+  forStorage?: boolean
+  outFields?: string
+  magicKey?: string
+  size?: number
+}
+
 type BaseQuery = {
-  clientId?: string,
-  clientSecret?: string,
+  boundary?: Boundary
+  clientId?: string
+  clientSecret?: string
+  focusPoint?: Point
+  text?: string
```

Comment:
Please remove `boundary`, `focusPoint` and `text` since they are not used in every public function call.